src/modules/saga/roomSaga.js

```diff
@@ -0,0 +1,141 @@
+import { call, takeEvery, put } from "redux-saga/effects";
+import axios from "axios";
+import { roomSliceActions } from "../slice/roomSlice";
+
+const getRoomSaga = function* () {
+  try {
+    const response = yield call(() =>
+      axios.get("http://localhost:4000/rooms", {
+        withCredentials: true,
+      })
+    );
+
+    if (response.data.message) {
+      yield put(roomSliceActions.getRoomFailure(response.data.rooms));
+      return;
+    }
+
+    yield put(roomSliceActions.getRoomsSuccess(response.data.rooms));
+  } catch (error) {
+    yield put(roomSliceActions.getRoomFailure(error));
+  }
+};
+
+const getNextRoomSaga = function* ({ payload }) {
+  try {
+    const response = yield call(() =>
+      axios.post(
+        "http://localhost:4000/rooms",
+        {
+          room: payload[5],
+          direction: "next",
+        },
+        {
+          withCredentials: true,
+        }
+      )
+    );
+
+    if (response.data.message) {
+      yield put(roomSliceActions.getRoomFailure(response.data));
+      return;
+    }
+
+    yield put(roomSliceActions.getRoomsSuccess(response.data.rooms));
+  } catch (error) {
+    yield put(roomSliceActions.getRoomFailure(error));
+  }
+};
+
+const getPrevRoomSaga = function* ({ payload }) {
+  try {
+    const response = yield call(() =>
+      axios.post(
+        "http://localhost:4000/rooms",
+        {
+          room: payload[5],
+          direction: "prev",
+        },
+        {
+          withCredentials: true,
+        }
+      )
+    );
+
+    if (response.data.message) {
+      yield put(roomSliceActions.getRoomFailure(response));
+      return;
+    }
+
+    yield put(roomSliceActions.getRoomsSuccess(response.data.rooms));
+  } catch (error) {
+    yield put(roomSliceActions.getRoomFailure(error));
+  }
+};
+
+const getRefreshRoomSaga = function* ({ payload }) {
+  try {
+    const response = yield call(() =>
+      axios.post(
+        "http://localhost:4000/rooms/refresh",
+        {
+          roomList: payload,
+        },
+        {
+          withCredentials: true,
+        }
+      )
+    );
+
+    if (response.data.message) {
+      yield put(roomSliceActions.getRoomFailure(response));
+      return;
+    }
+
+    yield put(roomSliceActions.getRoomsSuccess(response.data.rooms));
+  } catch (error) {
+    yield put(roomSliceActions.getRoomFailure(error));
+  }
+};
+
+const createRoomSaga = function* (action) {
```

Comment:
action을 받을 때 구조분해해서 {payload} 로 받는 것으로 통일하면 좋을 것 같습니다!
```suggestion
const createRoomSaga = function* ({payload}) {
  try {
    yield call(async () => {
      await axios.post(
        "http://localhost:4000/rooms/new",
        {
          roomData: payload,
        },
        {
          withCredentials: true,
        }
```